packages/grafana-prometheus/src/querybuilder/components/MetricCombobox.tsx

```diff
@@ -171,3 +179,35 @@ const formatKeyValueStringsForLabelValuesQuery = (query: string, labelsFilters?:
 
   return formatPrometheusLabelFiltersToString(queryString, labelsFilters);
 };
+
+const getMectricComboboxStyles = (theme: GrafanaTheme2, inlineFieldWidth: number, buttonSize: ComponentSize) => {
+  const buttonSizeProp = () => {
+    switch (buttonSize) {
+      case 'sm':
+        return 24;
+      case 'lg':
+        return 48;
+      default:
+        return 32;
+    }
+  };
+  const widthToSubstract = inlineFieldWidth * 8 + buttonSizeProp();
+  return {
+    adaptToParentQueryEditor: css({
+      // Take metrics explorer button into account
+      maxWidth: `calc(100% - 32px)`,
```

Comment:
Can we use a constant for the 32px?

Reply:
I finally get that value from the button height ([here](https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/Forms/commonStyles.ts)). Does this make sense or do you prefer the constant?